src/main/java/seedu/address/storage/JsonAdaptedTransaction.java

```diff
@@ -91,8 +91,11 @@ public Transaction toModelType() throws IllegalValueException {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                     Date.class.getSimpleName()));
         }
-        if (!Date.isValidDate(date)) {
-            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS);
+        if (!Date.isValidDateFormat(date)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS_WRONG_FORMAT);
+        }
+        if (!Date.isValidDateInput(date)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS_WRONG_DATE);
         }
```

Comment:
What's this? Aren't the first and second if condition the same, which means the second if statement will never run? 🤔